### PR TITLE
chore(deps)!: upgrade svgo to v3

### PR DESCRIPTION
**Pull request by dargmuesli**
### Summary

Upgrade dependency svgo to v3 and increase the minimum supported node version to v14.

### Motivation

Keep up to date with the svgo upstream library.

### Does this pull request introduce a breaking change?

- [x] Yes
- [ ] No

### Checklist

- [x] I have followed the [conventional commits](https://www.conventionalcommits.org/) standard for my commit messages
- [x] I have read the [contributing guidelines](https://github.com/cssnano/cssnano/blob/master/CONTRIBUTING.md)
- [x] I have searched/checked for existing issues and PRs

### Documentation
Please check if any of these apply:

- [ ] My proposed change require an update to the documentation
- [ ] I have added or updated the relevant documentation

### CODE REVIEW

**Comment by dargmuesli** [comment]
This PR is meant to get work started to support svgo v3 and thus in draft state. I'm not sure what else needs to be done as I'm new to this library's source. Feel free to add or suggest necessary changes. I'm open to improvement suggestions!

File: packages/postcss-svgo/package.json [hunk 1]
```diff
@@ -28,16 +28,15 @@
...
     "postcss-value-parser": "^4.2.0",
-    "svgo": "^2.7.0"
+    "svgo": "^3.0.0"
   },
   "bugs": {
     "url": "https://github.com/cssnano/cssnano/issues"
   },
   "engines": {
-    "node": "^10 || ^12 || >=14.0"
+    "node": ">=14.0"
```
Comment (dargmuesli) [inline packages/postcss-svgo/package.json:37]: https://github.com/svg/svgo/releases/tag/v3.0.0 lists this as a breaking change.

File: packages/postcss-svgo/package.json [hunk 2]
```diff
@@ -28,16 +28,15 @@
...
   },
   "devDependencies": {
-    "@types/svgo": "^2.6.2",
```
Comment (dargmuesli) [inline packages/postcss-svgo/package.json:40]: Types are now included in the base library.

File: packages/postcss-svgo/src/index.js [hunk 3]
```diff
@@ -43,12 +43,9 @@ function minifySVG(input, opts) {
   svg = svg.replace(escapedQuotes, '$1="$2"');
 
   const result = optimize(svg, opts);
-  if (result.error) {
```
Comment (dargmuesli) [inline packages/postcss-svgo/src/index.js:46]: All errors are now thrown. I think this makes this error check unnecessary.

File: tsconfig.json [hunk 4]
```diff
@@ -12,7 +12,8 @@
     "newLine": "lf",
     "allowSyntheticDefaultImports": true,
     "forceConsistentCasingInFileNames": true,
-    "strict": true
+    "strict": true,
+    "skipLibCheck": true
```
Comment (dargmuesli) [inline tsconfig.json:16]: Without this option, the `tsc` command would complain about typing inside `.ts` files inside the `node_modules` directory.

**Comment by CharleeWa** [comment]
Thank you for this PR, I found the questions in the Nuxtjs.

When I add some lib on the Nuxt projects, The command line print this warning:

`stable has been deprecated`.

So, If this PR is merged, Can I give a PR to Nuxt.js?  

Thank you so much!

**Comment by dargmuesli** [comment]
Hey @CharleeWa, thank you for your support.
I'm afraid, I don't know what "questions" and "some lib" is referring to. I also don't know where the warning originates from. Would you create a [minimal reproduction](https://stackblitz.com/github/nuxt/starter/tree/v3-stackblitz)?

Regarding Nuxt: `cssnano` is a subdependency of `@nuxt/vite-builder`, which is a subdependency of `nuxt`. If this PR is merged as a breaking change, another PR would be required for `@nuxt/vite-builder`.

**Review by ludofischer — COMMENTED** [review]
Thank you for your PR! I think the only issue is that we would need to release a major version of postcss-svgo, cssnano, cssnano-preset-default and cssnano-preset-advanced. Updating to a major version can be inconvenient for those users who install cssnano as a transitive dependency. I've done a quick check :
1. `css-minimizer-webpack-plugin` (4371000 downloads) already requires Node.js 14.15, but the most downloaded version is still the 3.x which only requires Node.js 12.x 
2. Parcel: only requires Node.js 12.x but I think they use lightningcss by default
3. vue-cli: Only requires Node.js 12.x, but is deprecated

What do you think @alexander-akait ? Can we drop Node < 14 and release cssnano 6 in the near future? If we release 6 I would at least also get rid of the YAML parser, since it's only necessary for the (udocumented) YAML configuration feature which I think nobody is using (if they do they can just turn it into JSON)

**Comment by ludofischer** [comment]
The postcss-svgo tests need to be updated because the result is slightly different but it looks equivalent (different attribute order and minified identifiers)

**Comment by alexander-akait** [comment]
@ludofischer I think it is time to drop old nodes, so I am fine with the next v6

**Comment by CharleeWa** [comment]
@dargmuesli 

Sorry, I wasn't specific enough. If you use the pnpm package manager, the installation dependency will have this prompt. Of course it's not a big deal, just a warning, but it doesn't look very comfortable. So, I think we should wait for cssnano to release a new version, nuxt will update cssnano's version, and the problem will not exist.

截屏2022-10-25 13 58 01

**Comment by ludofischer** [comment]
Thank you for updating the tests. It might take a bit longer before we release 6.x  New cssnano releases take a long time to reach users, so I would like to do at least another 5.x release with the updates we've just merged on master. Maybe it's also possible to solve some merge-rules bugs on 5.x before we switch to 6.  
I'm also thinking we could use the 6 release to rewrite the configuration loading and parsing (see https://github.com/cssnano/cssnano/issues/1436 for an example where the current behaviour is a bit unintuitive).

File: tsconfig.json (hunk 4 above)
Comment (TrySound) [reply tsconfig.json:16]: This is fixed in https://github.com/svg/svgo/releases/tag/v3.0.1

Comment (dargmuesli) [reply tsconfig.json:16]: Thank you for the notice!
It now gives
```
node_modules/.pnpm/svgo@3.0.1/node_modules/svgo/plugins/plugins-types.ts:1:28 - error TS7016: Could not find a declaration file for module 'csso'. '/mnt/data/WorkingDirectory/Git/dargmuesli/cssnano/node_modules/.pnpm/csso@5.0.5/node_modules/csso/cjs/index.cjs' implicitly has an 'any' type.
  Try `npm i --save-dev @types/csso` if it exists or add a new declaration (.d.ts) file containing `declare module 'csso';`

1 import type * as csso from 'csso';            
```
Is that something that should be fixed by cssnano or svgo?

Comment (TrySound) [reply tsconfig.json:16]: You need to add `@types/csso` to dev dependencies

**Comment by dargmuesli** [comment]
I updated to v3.0.1 and reverted the `skipLibCheck` addition.

**Comment by alexander-akait** [comment]
@dargmuesli can you rebase to solve pnpm merge

**Comment by dargmuesli** [comment]
Rebased as requested ✅

**Comment by zdm** [comment]
When this PR will be merged?

**Comment by silkfire** [comment]
Bump, when can we merge this PR? @ludofischer

**Comment by mahnunchik** [comment]
I'm looking forward to have it merged

**Comment by dargmuesli** [comment]
Please upvote the PR by giving a :+1: on the first comment here and do not add "me too" comments, thanks!

**Comment by zdm** [comment]
Oh, is is possible to merge it finally?
PLEASE.

File: packages/postcss-svgo/package.json [hunk 5]
```diff
@@ -28,16 +28,16 @@
   "repository": "cssnano/cssnano",
   "dependencies": {
     "postcss-value-parser": "^4.2.0",
-    "svgo": "^2.7.0"
+    "svgo": "^3.0.1"
```
Comment (VIKTORVAV99) [inline packages/postcss-svgo/package.json:31]: If we bump this to 3.0.2 we can get rid of the `@types/csso` package as well.
```suggestion
    "svgo": "^3.0.2"
```

File: packages/postcss-svgo/package.json [hunk 6]
```diff
@@ -28,16 +28,16 @@
...
   "bugs": {
     "url": "https://github.com/cssnano/cssnano/issues"
   },
   "engines": {
-    "node": "^10 || ^12 || >=14.0"
+    "node": "^14 || ^16 || >= 18"
   },
   "devDependencies": {
-    "@types/svgo": "^2.6.2",
+    "@types/csso": "^5.0.0",
```
Comment (VIKTORVAV99) [inline packages/postcss-svgo/package.json:40]: This can be removed now as they are no longer needed since v3.0.2 of SVGO.

**Review by VIKTORVAV99 — COMMENTED** [review]
SVGO has released a new [patch version](https://github.com/svg/svgo/releases/tag/v3.0.2) so the csso types package should no longer be needed.

File: packages/postcss-svgo/package.json (hunk 6 above)
Comment (dargmuesli) [reply packages/postcss-svgo/package.json]: Removed :)

File: packages/postcss-svgo/package.json (hunk 5 above)
Comment (dargmuesli) [reply packages/postcss-svgo/package.json]: Updated! :rocket:

**Comment by eddiemonge** [comment]
Considering Node14 is almost out of maintenance schedule, getting this in sooner would be much appreciated. I get moving slow with major updates, but considering users _do not have to upgrade unless they want to_, doing a major update shouldn't take so long or be so painful.

**Comment by alexander-akait** [comment]
@ludofischer Maybe it is time to do major release and drop old nodejs support?

**Comment by ludofischer** [comment]
What features would you want from SVGO 3? There seem to be at least one unsolved regression from SVGO 2: https://github.com/svg/svgo/issues/1746. I don't know how impactful these are though.
I was hoping to find the time to improve the configuration reading as I think some configuration options right now have no effect (as that would be another breaking change), but obviously it seems I never manage to get around to doing it. So we might as well release cssnano 6.0 just with SVGO bump, dropping the YAML support and possibly dropping normalize-url.

**Review by ludofischer — APPROVED** [review]
I think we can merge this.

**Pull request merged**